src/base/ValueVector.h

```diff
@@ -1,27 +1,75 @@
 #pragma once
 
-#include <boost/ptr_container/ptr_vector.hpp>
+#include <vector>
+#include <memory>
+#include <boost/iterator/indirect_iterator.hpp>
+
 
 namespace NuTo
 {
 
 //! @brief container that stores values of T and keeps references to these values valid
+//! @remark Normally, it is a no-brainer to use boost::ptr_vector here. It provides all
+//! the nice value semantics for free. However, there is an issue regarding move.
+//!         https://stackoverflow.com/questions/44130076/moving-a-boostptr-vector
+//! Therefore, here is a custom implementation with only the basic features.
 template <typename T>
-class ValueVector : public boost::ptr_vector<T>
+class ValueVector
 {
-public:
+    using Data = std::vector<std::unique_ptr<T>>;
+    //! @brief indirect (dereferencing) iterator to provide value semantics for the iterators
+    typedef boost::indirect_iterator<typename Data::iterator> IndirectIterator;
+    typedef boost::indirect_iterator<typename Data::const_iterator> ConstIndirectIterator;
 
+public:
     template <typename... TArgs>
     T& Add(TArgs&&... args)
     {
-        this->push_back(new T(std::forward<TArgs>(args)...));
-        return this->back();
+        mData.push_back(std::make_unique<T>(std::forward<TArgs>(args)...));
+        return *mData.back();
```

Comment:
I'm uncertain about the operator precedence is such cases (obviously this is correct), so I'd prefer `return *(mData.back());`. What do you think?